client/rpc-api/Cargo.toml

```diff
@@ -16,7 +16,7 @@ targets = ["x86_64-unknown-linux-gnu"]
 codec = { package = "parity-scale-codec", version = "1.3.4" }
 derive_more = "0.99.2"
 futures = { version = "0.3.1", features = ["compat"] }
-jsonrpc-core = "15.1.0"
+jsonrpc-core = { version = "15.1.0", features = ["arbitrary_precision"] }
```

Comment:
I think this would crash the node.

@tomusdrw am I correct or do I mix some stuff up here?

Reply:
AFAIR it breaks some other features that we heavily depend on (like `flatten`) in some cases. To overcome `u128` serialization, serde annotations should be used like we do  here:
https://github.com/paritytech/substrate/blob/4320a52d3a6604839751b5776dd0f174f0405438/frame/transaction-payment/rpc/runtime-api/src/lib.rs#L40-L43

Reply:
I don't think that would work here as I use serde_json directly and not struct. See [here](https://github.com/paritytech/substrate/blob/ca23c7a/client/telemetry/src/lib.rs#L317).

One alternative would be to allow a string for the one (maybe 2) cases where we use a u128. But then the telemetry server will need to be able to handle it. The u128 comes from [here](https://github.com/paritytech/substrate/blob/ca23c7ae1c17691a7d75ef997373f8a62da5c92f/client/service/src/builder.rs#L715) (it's possible there is another one somewhere).

Reply:
> AFAIR it breaks some other features that we heavily depend on (like flatten) in some cases. But this should be covered by tests.

(A lot of tests broke but I fixed them all)

Reply:
We opted to stringify the timestamp that was in u128 instead